File: javademo/src/main/java/com/awesome/javademo/activitys/MainActivity.java
package com.awesome.javademo.activitys;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.awesome.javademo.R;
import com.awesome.javademo.abstracts.AbstractAnimal;
import com.awesome.javademo.abstracts.Animal;
import com.awesome.javademo.abstracts.Panda;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        AbstractAnimal abstractAnimal = new Animal();//向上转型，调用子类的重写方法，或父类未被重写的方法
        abstractAnimal.eat();
        Animal a = (Animal) abstractAnimal;// 向下转型，可以调用子类的重写方法，和子类独有的方法。即子类的全部方法。
        a.eat();
        a.sleep();
    }
}

File: kotlindemo/src/main/java/com/awesome/kotlindemo/MainActivity.kt
package com.awesome.kotlindemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.awesome.sdk.util.ShowLogUtil
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        InitOrderDemo("Jane")

//        Beginner().main()
//        var customer = Customer("Jane", "23234@qq.com")
//        // 关键字with代码块可以直接调用该实例的方法
//        with(customer){
//            foo()
//            ShowLogUtil.info(getString(1))
//            ShowLogUtil.info(tryCatch())
//        }
//
//        // 配置对象的属性
//        Beginner().apply {
//            x = 9
//        }
    }

//        Fruit();
//        Fruit("violet");
//        Fruit("violet","Grape");
//        Grape("violet")
//        Grape("violet", "Grape")
//        Grape("Shandong");
//        Grape("Shandong","hugePeak");
//        super.getDelegate()
//        AdvanceLayout(this);
}